### ISSUE TITLE
Database upgrade is not performed properly

### ISSUE THREAD

**Issue by tomchiverton** [Cacti/plugin_thold#250]
Using latest from git:

```
2019/02/16  22:52:25 - CMDPHP PHP ERROR NOTICE Backtrace:   (/plugins/thold/thold_graph.php[55]:tholds(),  /plugins/thold/thold_graph.php[482]:CactiErrorHandler())
--
2019/02/16 22:52:25 - ERROR PHP NOTICE in   Plugin 'thold': Undefined index: acknowledgment in file:  /usr/share/cacti/plugins/thold/thold_graph.php  on line: 482

```

**Comment by netniV** [comment]
This error occurs before you visit the thold.php or plugins.php page to apply the plugin upgrade.  To be sure, please let us know what version is shown against your plugin and then run the following SQL:

```sql
desc thold_data;
```
or
```sql
show create table\G
```

**Comment by netniV** [comment]
This has now been corrected in the latest development version.

**Comment by tomchiverton** [comment]
Field	Type	Null	Key	Default	Extra
id	int(11)	NO	PRI	(null)	auto_increment
name	varchar(150)	YES		(null)	
name_cache	varchar(100)	NO			
local_data_id	int(11) unsigned	NO	MUL	0	
data_template_rrd_id	int(11) unsigned	NO	MUL	0	
local_graph_id	int(11) unsigned	NO	MUL	0	
graph_template_id	int(11) unsigned	NO	MUL	0	
data_template_id	int(11) unsigned	NO	MUL	0	
data_source_name	varchar(100)	NO			
thold_hi	varchar(100)	YES		(null)	
thold_low	varchar(100)	YES		(null)	
thold_fail_trigger	int(10) unsigned	YES		(null)	
time_hi	varchar(100)	YES		(null)	
time_low	varchar(100)	YES		(null)	
time_fail_trigger	int(12)	NO		1	
time_fail_length	int(12)	NO		1	
thold_fail_count	int(11)	NO		0	
thold_alert	int(1)	NO		0	
prev_thold_alert	int(1)	NO		0	
thold_enabled	enum('on','off')	NO	MUL	on	
thold_type	int(3)	NO		0	
bl_enabled	enum('on','off')	NO		off	
bl_ref_time_range	int(10) unsigned	YES		(null)	
bl_pct_down	varchar(100)	YES		(null)	
bl_pct_up	varchar(100)	YES		(null)	
bl_fail_trigger	int(10) unsigned	YES		(null)	
bl_fail_count	int(11) unsigned	YES		(null)	
bl_alert	int(2)	NO		0	
lastread	varchar(100)	YES		(null)	
lasttime	timestamp	NO		current_timestamp()	on update current_timestamp()
oldvalue	varchar(100)	NO			
repeat_alert	int(10) unsigned	YES		(null)	
notify_extra	varchar(512)	YES			
data_type	int(12)	NO		0	
host_id	int(10)	YES		(null)	
syslog_priority	int(2)	YES		3	
syslog_facility	int(2)	YES		(null)	
syslog_enabled	char(3)	NO			
cdef	int(11)	NO		0	
percent_ds	varchar(64)	NO		0	
expression	varchar(512)	NO			
thold_template_id	int(11) unsigned	NO	MUL	0	
template_enabled	varchar(3)	NO	MUL		
tcheck	int(1)	NO	MUL	0	
exempt	char(3)	NO			
acknowledgement	char(3)	NO			
thold_hrule_alert	int(11) unsigned	YES		(null)	
thold_hrule_warning	int(11) unsigned	YES		(null)	
restored_alert	char(3)	NO			
reset_ack	char(3)	NO			
persist_ack	char(3)	NO			
email_body	varchar(1024)	YES			
email_body_warn	varchar(1024)	NO			
trigger_cmd_high	varchar(512)	NO			
trigger_cmd_low	varchar(512)	NO			
trigger_cmd_norm	varchar(512)	NO			
bl_thold_valid	int(10) unsigned	NO		0	
time_warning_fail_length	int(12)	NO		1	
thold_warning_hi	varchar(100)	YES		(null)	
thold_warning_low	varchar(100)	YES		(null)	
thold_warning_fail_trigger	int(10) unsigned	YES		(null)	
thold_warning_fail_count	int(11)	NO		0	
time_warning_hi	varchar(100)	YES		(null)	
time_warning_low	varchar(100)	YES		(null)	
time_warning_fail_trigger	int(12)	NO		1	
notify_warning_extra	mediumtext	YES		(null)	
notify_alert	int(10) unsigned	NO		1	
notify_warning	int(10) unsigned	NO		1	
snmp_event_category	varchar(255)	YES		(null)	
snmp_event_severity	tinyint(1)	NO		3	
snmp_event_warning_severity	tinyint(1)	NO		2	
thold_daemon_pid	varchar(25)	NO	MUL		
notes	varchar(1024)	YES

**Comment by tomchiverton** [comment]
I git pull'd so it's all latest
```

# git status
On branch develop
Your branch is up to date with 'origin/develop'.

nothing to commit, working tree clean
#
```

http://monitor.falkensweb.com/plugins/thold/thold.php no fails to load at all.

`
2019/02/27 20:54:10 - CMDPHP ERRORS DETECTED - DISABLING PLUGIN 'thold'
2019/02/27 20:54:10 - CMDPHP PHP ERROR Backtrace:  (CactiShutdownHandler())
2019/02/27 20:54:10 - ERROR PHP ERROR in   Plugin 'thold': Uncaught Error: Call to undefined function  get_daysfromtime() in  /usr/share/cacti/plugins/thold/thold_functions.php:87 Stack trace: #0  /usr/share/cacti/plugins/thold/thold.php(988):  get_time_since_last_event(Array) #1  /usr/share/cacti/plugins/thold/thold.php(141): list_tholds() #2 {main}  thrown in file: /usr/share/cacti/plugins/thold/thold_functions.php on  line: 87
`

**Comment by netniV** [comment]
See closed issue #261
